shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/output/DefaultOutputProcessing.java

```diff
@@ -103,81 +100,90 @@ public void manageTarGzResult(File in, File parent, CarminDatasetProcessing proc
 				IOUtils.copy(fin, Files.newOutputStream(currentFile.toPath()));
 
 				if (parsedEntry.endsWith(this.resultFileName)) {
-
-					LOG.info("Processing result JSON [{}]...", parsedEntry);
-					try {
-						inputDatasets = this.getDatasetsFromResultJSON(currentFile);
-
-						if(inputDatasets.isEmpty()){
-							LOG.warn("No datasets found in result JSON [{}].", parsedEntry);
-						}
-
-					} catch (Exception e) {
-						LOG.error("Could not read result JSON file", e);
-					}
-
+					resultJson = currentFile;
 				} else {
 					// For all other files that are not a result.json or a folder, create a processed dataset and a dataset processing
 					outputFiles.add(currentFile);
+					LOG.info("Output file [{}] found in archive.", parsedEntry);
 				}
 			}
 
+			List<Dataset> inputDatasets = this.getInputDatasets(resultJson, in.getName());
+
+			if(inputDatasets.isEmpty()) {
+				throw new Exception("No input datasets found.");
+			}
+
 			if(outputFiles.isEmpty()){
-				LOG.warn("No processable file found in Tar result.");
+				throw new Exception("No processable file found in Tar result.");
 			}
 
 			this.createProcessedDatasets(outputFiles, processing, inputDatasets);
 
 			this.deleteCacheDir(Paths.get(cacheFolder.getAbsolutePath()));
 
 		} catch (Exception e) {
-			LOG.error("An error occured while extracting result from tar.gz file: ", e);
+			LOG.error("An error occured while extracting result from result archive.", e);
+			importerService.createFailedJob(in.getPath());
 		}
 	}
 
 
-	private List<Dataset> getDatasetsFromResultJSON(File resultJson) throws IOException, JSONException {
-
-		InputStream is = new FileInputStream(resultJson);
-		JSONObject json = new JSONObject(IOUtils.toString(is, StandardCharsets.UTF_8));
+	private List<Dataset> getInputDatasets(File resultJson, String tarName) throws IOException, JSONException {
 
 		List<Long> datasetIds = new ArrayList<>();
+		List<String> candidates = new ArrayList<>();
 
-		List<String> filenames = new ArrayList<>();
+		candidates.add(tarName);
 
-		if(!json.has(JSON_INFILE)){
-			LOG.error("No key [{}] found in [{}]", JSON_INFILE, resultJson.getAbsolutePath());
-			return new ArrayList<>();
-		}
-
-		Object infiles = json.get(JSON_INFILE);
-
-		if (infiles instanceof JSONArray) {
-			// case "["id+XXX+filename.nii", "YYY+filename.nii"]"
-			JSONArray array = (JSONArray) infiles;
-			for (int i=0 ; i < array.length(); i++) {
-				filenames.add(array.getString(i));
-			}
+		if (resultJson == null) {
+			LOG.info("No result JSON found in archive.");
 		} else {
-			// Case "id+XXX+filename.nii"
-			String value = (String) infiles;
-			filenames.add(value);
+			LOG.info("Processing result JSON [{}]...", resultJson.getName());
+
+			InputStream is = new FileInputStream(resultJson);
```

Comment:
I think you may need a try with ressource here to close properly the stream at the end.